packages/remark-shaku-code-annotate/src/index.ts

```diff
@@ -184,11 +199,14 @@ export const remarkShakuCodeAnnotate = (
             shouldHighlighNextSourceLine = false;
             const shouldDim = isDimBlock || shouldDimNextSourceLine;
             shouldDimNextSourceLine = false;
+            const shouldFocus = isFocusBlock || shouldFocusNextSourceLine;
+            shouldFocusNextSourceLine = false;
             const sourceLine = line.line;
-            const prefix = `<div class="line${
-              shouldHighlight ? " highlight" : ""
-            }${shouldDim ? " dim" : ""}">`;
-
+            const focusFlag = hasFocusElement(lines);
```

Comment:
this check is run for every line, which is not performant.
We can move it to outer level and only check it once

Reply:
Thank you for pointing out this ! It has been fixed.